okio-kotlin/src/main/java/okio/HashingSink.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("NOTHING_TO_INLINE") // Aliases to public API.
+
+package okio
+
+/**
+ * Returns a [HashingSink] that uses the obsolete MD5 hash algorithm to produce 128-bit hashes.
+ *
+ * @see HashingSink.md5
+ */
+inline fun Sink.hashMd5(): HashingSink = HashingSink.md5(this)
+
+/**
+ * Returns a [HashingSink] that uses the obsolete SHA-1 hash algorithm to produce 160-bit hashes.
+ *
+ * @see HashingSink.sha1
+ */
+inline fun Sink.hashSha1(): HashingSink = HashingSink.sha1(this)
+
+/**
+ * Returns a [HashingSink] that uses the SHA-256 algorithm to produce 256-bit hashes.
+ *
+ * @see HashingSink.sha256
+ */
+inline fun Sink.hashSha256(): HashingSink = HashingSink.sha256(this)
+
+/**
+ * Returns a [HashingSink] that uses the SHA-512 algorithm to produce 512-bit hashes.
+ *
+ * @see HashingSink.sha512
+ */
+inline fun Sink.hashSha512(): HashingSink = HashingSink.sha512(this)
+
+/**
+ * Returns a [HashingSink] that uses the obsolete SHA-1 HMAC algorithm with [key] to produce 160-bit
+ * hashes.
+ *
+ * @see HashingSink.hmacSha1
+ */
+inline fun Sink.hashHmacSha1(key: ByteString): HashingSink = HashingSink.hmacSha1(this, key)
```

Comment:
Should `key` become a nullable `hmacKey` parameter on `hashSha1()` with a default of `null`? Then it's one function that switches between two impls rather than two functions.

Reply:
I think the use cases are pretty different.

Reply:
Also: omit the word hash here? It's like ATM machine or ABS brakes.

Reply:
Just for the hmac ones or for all of them?

The names are interesting for these overloads because the return type is significant here. You have to retain the `HashinkSink`/`HashingSource` in order to grab the hash. Should these names be `hashingSha1()`? What does that do to the hmac names then?

Reply:
sha1Sink? Tough call. I don’t think these things get much use.

Reply:
I think we want:
sha1() –> returns ByteString
hmacSha1(ByteString) -> returns ByteString
sha1Sink() –> returns HashingSink

Reply:
The only thing I don't like about that is that it creates a disparity for things like `.gzip()`.